config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/OAuth2Configurer.java

```diff
@@ -96,7 +97,8 @@ private void initClientConfigurer() {
 	}
 
 	private void initResourceServerConfigurer() {
-		this.resourceServerConfigurer = new OAuth2ResourceServerConfigurer<>();
+		ApplicationContext context = getBuilder().getSharedObject(ApplicationContext.class);
+		this.resourceServerConfigurer = new OAuth2ResourceServerConfigurer<>(context);
```

Comment:
@jgrandja you and I were talking about trying to keep the ApplicationContext resolution approach consistent across OAuth2 configurers; however with this set of tests for bearer token resolver, it still seemed the cleanest approach. Let me know if you see one you'd prefer more.

Reply:
For `oauth2-client` we use `OAuth2ClientConfigurerUtils` to get `@Bean`'s from the context - passing in the builder (http). This centralizes the logic for `@Bean` lookup so it can be reused across configurer's since there is more than one for `oauth2-client`. I'm not sure if there will be more than one for resource server. Either way, I think it's fine leaving as is.